docs/docs/reference/gen_notebooks/feedback_prod.md

```diff
@@ -0,0 +1,179 @@
+---
+title: Log Feedback from Production
+---
+
+
+:::tip[This is a notebook]
+
+<a href="https://colab.research.google.com/github/wandb/weave/blob/master/docs/./notebooks/feedback_prod.ipynb" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link button button--secondary button--med margin-right--sm notebook-cta-button"><div><img src="https://upload.wikimedia.org/wikipedia/commons/archive/d/d0/20221103151430%21Google_Colaboratory_SVG_Logo.svg" alt="Open In Colab" height="20px" /><div>Open in Colab</div></div></a>
+
+<a href="https://github.com/wandb/weave/blob/master/docs/./notebooks/feedback_prod.ipynb" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link button button--secondary button--med margin-right--sm notebook-cta-button"><div><img src="https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg" alt="View in Github" height="15px" /><div>View in Github</div></div></a>
+
+:::
+
+
+
+<img src="http://wandb.me/logo-im-png" width="400" alt="Weights & Biases" />
+<!--- @wandbcode{feedback-colab} -->
+
+
+It is often hard to automatically evaluate a generated LLM response so, depending on your risk tolerance, you can gather direct user feedback to use to find areas to improve.
+
+In this tutorial, we'll use a custom RAG chatbot as an example app with which the users can interact and which allows us to collect user feedback.
+We'll use Streamlit to build the interface and we'll capture the LLM interactions and feedback in Weave.
+
+## Setup
+
+
+```python
+!pip install weave openai streamlit
+```
+
+First, create a file called `secrets.toml` and add an OpenAI key so it works with `[st.secrets](https://docs.streamlit.io/develop/api-reference/connections/st.secrets)`. You can [sign up](https://platform.openai.com/signup) on the OpenAI platform to get your own API key. 
+
+
+```python
+# secrets.toml
+OPENAI_API_KEY = "your OpenAI key"
+```
+
+Next, create a file called `chatbot.py` with the following contents:
+
+
+```python
+# chatbot.py
+
+import weave
+from openai import OpenAI
+import streamlit as st
+
+st.title("Add feedback")
+
+# highlight-next-line
+@weave.op
+def chat_response(prompt):
+    stream = client.chat.completions.create(
```

Comment:
If we don't put in an actual model from Weave we should defintely reference to another upcoming cookbook or docs page for how to deploy models from Weave (to close the loop from experimentation to production).

For #2071  I'm using something like this: 

```python
@weave.op()
def production_chat_response(model_url: str, prompt: str):
    """
    The main model function that will take the prompt and the model_url to generate a response.
    Optionally, this function call also be decorated with @weave.op(). 
    """
    
    if model_url.startswith("weave://"):
        # option 1 - get the model object from weave
        RagModel = get_weave_model(model_url)
        
        # NOTE: we use the .call object so that the call is returned and feedback can be added
        # TODO: check answer from Adam if i still need to a the specific model object in the call
        #data, call = asyncio.run(RagModel.predict.call(query=prompt, n_documents=2))  
        data = asyncio.run(RagModel.predict(prompt))   
        result_content = data['result']['content']
        call = weave.get_current_call()  

    elif model_url.startswith("http"):
        # option 2 - the model is already served to a certain endpoint
        response = requests.post(
            model_url,
            json={
                "query": prompt,
                "n_documents": 2,
            },
        )
        data = response.json()
        result_content = data['result']['result']['content']

        # TODO: this works when this is a weave.op - how to get the last call otherwise?
        call = weave.get_current_call() 
    else:
        print("Please pass in a URL to a weave object or http endpoint.")
 ```